docs/database/database-spans.md

```diff
@@ -268,6 +302,98 @@ Placeholders in a parameterized query SHOULD not be sanitized. E.g. `where id =
 e.g. from `IN (?, ?, ?, ?)` to `IN (?)`, as this can help with extremely long IN-clauses,
 and can help control cardinality for users who choose to (optionally) add `db.query.text` to their metric attributes.
 
+## Generating a summary of the query text
+
+The `db.query.summary` attribute captures a shortened representation of a query text
+which SHOULD have low-cardinality and SHOULD NOT contain any dynamic or sensitive data.
+
+> [!NOTE]
+> The `db.query.text` attribute is intended to identify individual queries. Even though
+> it is sanitized if captured by default, it could still have high cardinality and
+> might reach hundreds of lines of code.
+>
+> The `db.query.summary` is intended to provide a less granular grouping key that
+> can be used as a span name or a metric attribute in common cases. It SHOULD
+> only contain information that has a significant impact on the query, database,
+> or application performance.
+
+Instrumentations that support query parsing SHOULD generate query summary when
+one is not readily available from other sources.
+
+The summary SHOULD preserve the following parts of query in the order they were provided:
+
+- operations such as SQL SELECT, INSERT, UPDATE, DELETE or other commands
+- operation targets such as collection or database names
+
+Instrumentation MAY include additional details such as specific SQL clauses as long
+as summary remains relatively short and its cardinality remains low comparing to
+the `db.query.text`.
+
+The instrumentation SHOULD parse the query and extract a list of operations and
+corresponding targets from the query. It SHOULD set `db.query.summary` attribute
+to the value formatted in the following way:
+
+```
+{operation1} {target1} {operation2} {target2} {target3} ...
+````
+
+Instrumentations SHOULD capture the values of operations and targets as provided
+by the application without attempting to do any case normalization. If the operation
+and target value is populated on `db.operation.name`, `db.collection.name`, `db.namespace`
+or other attributes, it SHOULD match the value used in the `db.query.summary`.
```

Comment:
I think we've defined `db.namespace` in terms of the connection now, so not sure if it would ever be extracted from the query

Reply:
I think we still allow instrumentations to keep track if the current connection

> A connection's currently associated database may change during its lifetime, e.g. from executing `USE <database>`.

> If instrumentation is unable to capture the connection's currently associated database on each query without triggering an additional query to be executed (e.g. `SELECT DATABASE()`),
then it is RECOMMENDED to fallback and use the database provided when the connection was established.

>  Instrumentation SHOULD document if `db.namespace` reflects the database provided when the connection was established.

So it's technically possible if they match

Reply:
removed `db.namespace` from this list, it's not necessary there anyway

Reply:
I think I see...

so someone is using database `MyDB1` and then executes `USE MyDB2`:

* `db.query.text=USE MyDB`
* `db.query.summary=USE MyDB`
* `db.operation.name=USE`
* `db.collection.name` is not captured
* `db.namespace=MyDB1` (I think, b/c that was the namespace associated with the connection at the time?)

and internally if the instrumentation is tracking namespace changes, then on the next query `SELECT * FROM table5`:

* `db.query.text=SELECT * FROM table5`
* `db.query.summary=SELECT table5`
* `db.operation.name=SELECT`
* `db.collection.name=table5`
* `db.namespace=MyDB2`

Reply:
I think so.

Until recently we used to say that even the first span should point to the new database: https://github.com/open-telemetry/semantic-conventions/commit/25d438c3bf7234d27ba7a55e4520943e601f3cb9

> For commands that switch the keyspace, this SHOULD be set to the target keyspace (even if the command fails).

But the old version goes against of the current attribute definition.